pkg/network/http/http_statkeeper.go

```diff
@@ -31,30 +33,63 @@ func newHTTPStatkeeper(maxEntries int, telemetry *telemetry) *httpStatKeeper {
 }
 
 func (h *httpStatKeeper) Process(transactions []httpTX) {
-	var dropped int
 	for _, tx := range transactions {
-		key := h.newKey(tx)
-		stats, ok := h.stats[key]
-		if !ok && len(h.stats) >= h.maxEntries {
-			dropped++
+		if tx.Incomplete() {
+			h.handleIncomplete(tx)
 			continue
 		}
 
-		stats.AddRequest(tx.StatusClass(), tx.RequestLatency())
-		h.stats[key] = stats
+		h.add(tx)
 	}
 
-	atomic.AddInt64(&h.telemetry.dropped, int64(dropped))
 	atomic.StoreInt64(&h.telemetry.aggregations, int64(len(h.stats)))
 }
 
 func (h *httpStatKeeper) GetAndResetAllStats() map[Key]RequestStats {
 	ret := h.stats // No deep copy needed since `h.stats` gets reset
 	h.stats = make(map[Key]RequestStats)
+	h.incomplete = make(map[Key]httpTX)
 	h.interned = make(map[string]string)
 	return ret
 }
 
+func (h *httpStatKeeper) add(tx httpTX) {
+	key := h.newKey(tx)
+	stats, ok := h.stats[key]
+	if !ok && len(h.stats) >= h.maxEntries {
+		atomic.AddInt64(&h.telemetry.dropped, 1)
+		return
+	}
+
+	stats.AddRequest(tx.StatusClass(), tx.RequestLatency())
+	h.stats[key] = stats
+}
+
+func (h *httpStatKeeper) handleIncomplete(tx httpTX) {
```

Comment:
This could use a comment, so I took a stab at it:
```suggestion
// handleIncomplete deals with incomplete transactions, which contains only the request or response information,
// by merging the response into the request.
func (h *httpStatKeeper) handleIncomplete(tx httpTX) {
```

Reply:
Done!